frontends/p4/redundantParsers.cpp

```diff
@@ -37,6 +37,11 @@ bool FindRedundantParsers::preorder(const IR::P4Parser *parser) {
 }
 
 const IR::Node *EliminateSubparserCalls::postorder(IR::MethodCallStatement *mcs) {
+    // Only remove redundant parser applications in parser states. Parser applications
+    // made in illegal locations (such as control blocks) will be detected by and error
+    // will be reported in later passes (i.e. DiscoverInlining).
```

Comment:
It would be great if you also add test that covers this case.

Reply:
We already have test [issue1331.p4](https://github.com/p4lang/p4c/blob/main/testdata/p4_16_errors_outputs/issue1331.p4-stderr) that will fail without this change.